### PR TITLE
Documentation for the macros in `Assertions.jl`

### PR DESCRIPTION
**Pull request by StevellM**
This is a suggestion for documentations of the macros in the file `Assertions.jl`.
It is not perfect, and namings can be discussed, nothing is strict.

I have added also something on the online manual: I have not been able to render it on my local machine so I cannot say whether or not it "looks good". Same as before, the namings are not perfect neither definitive. I am open to change these.

I have also moved the export of `@vtime` to the correct file.

### CODE REVIEW

**Comment by StevellM** [comment]
This should also closed #919

File: docs/src/features/macros.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Macros
+```@meta
+CurrentModule = Hecke
+DocTestSetup = quote
+    using Hecke
+  end
+```
+
+We refer here to the extra macros available on Hecke.
```
Comment (fingolfin) [inline docs/src/features/macros.md:9]:
```suggestion
We describe here various macros provided by Hecke.
```

File: docs/src/features/macros.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+get_assert_level(s::Symbol)
+```
+
+### Check
+
+```@docs
+@hassert
+```
+
+## Misc
```
Comment (fingolfin) [inline docs/src/features/macros.md:57]:
```suggestion
## Miscellaneous
```

File: src/Assertions.jl [hunk 3]
```diff
@@ -89,6 +86,64 @@ function _global_indent()
...
+@doc Markdown.doc"""
+    @vprint(S::Symbol, k::Int, msg::String)
+    @vprint S k msg
+
+    @vprint(S::Symbol, msg::String)
+    @vprint S msg
+
+This macro can be used to control printings inside the code.
+
+The macro `vprint` has two or three entry arguments: a symbol `S` refered as a
```
Comment (fingolfin) [inline src/Assertions.jl:97]:
```suggestion
The macro `vprint` takes two or three arguments: a symbol `S` specifying a
```

File: src/Assertions.jl [hunk 4]
```diff
@@ -89,6 +86,64 @@ function _global_indent()
...
+*verbose scope*, an optional integer `k` and a string `msg`. If `k` is not
```
Comment (fingolfin) [inline src/Assertions.jl:98]: This is of course not your fault, but: "verbose scope" and "verbose level" IMHO really should be "verbosity scope" and "verbosity level". But of course that's something @thofma and @fieker should decide.

File: src/Assertions.jl [hunk 5]
```diff
@@ -89,6 +86,64 @@ function _global_indent()
...
+specified, it is set by default to $1$.
+
+To each verbose scope `S` is associated a *verbose level* `l` which is cached.
```
Comment (fingolfin) [inline src/Assertions.jl:101]:
```suggestion
To each verbose scope `S` is associated a *verbose level* `l`.
```
What do you mean by "is cached"? Maybe that it is persistent?

**Comment by thofma** [comment]
If you could rename them as @fingolfin suggested, e.g. `verbosity_scope` etc, this would be great.

**Comment by StevellM** [comment]
Sure, it was ready to go, I was waiting for extra approval as Max suggested.

File: src/Assertions.jl [hunk 6]
```diff
@@ -125,11 +235,61 @@ macro v_do(s, l::Int, action)
   end
 end
 
+@doc Markdown.doc"""
+    set_verbose_level(s::Symbol, l::Int) -> Int
```
Comment (fingolfin) [inline src/Assertions.jl:239]: Also rename this and other such functions? I.e.:
```suggestion
    set_verbosity_level(s::Symbol, l::Int) -> Int
```

File: src/Assertions.jl [hunk 7]
```diff
@@ -125,11 +235,61 @@ macro v_do(s, l::Int, action)
...
+`s` to `l`.
+
+One can access the current verbosity level of `s` by calling the function
+[`get_verbose_level`](@ref).
+
+If `s` is not yet known as a verbosity scope, the function raises an `ErrorException`
+showing the error message "Not a valid symbol". One can add `s` to the list of
+verbosity scopes by calling the function [`add_verbose_scope`](@ref).
+
+# Example
```
Comment (fingolfin) [inline src/Assertions.jl:251]: The convention (at least in Oscar, and Julia itself) is to always call this section "Examples", so:
```suggestion
# Examples
```

File: src/Assertions.jl [hunk 8]
```diff
@@ -196,11 +352,96 @@ function set_assert_level(s::Symbol, l::Int)
...
 end
 
+@doc Markdown.doc"""
+    @hassert(S::Symbol, k::Int, assert::Expr)
+    @hassert S k assert
+
+    @hassert(S::Symbol, assert::Expr)
+    @hassert S assert
+
+This macro can be used to control assertions check inside the code.
```
Comment (fingolfin) [inline src/Assertions.jl:395]:
```suggestion
This macro can be used to control assertion checks inside the code.
```

File: src/Assertions.jl (hunk 6 above)
Comment (StevellM) [reply src/Assertions.jl:239]: Well of course, I can also maybe change the `assert` to `assertion` in the similar way, if I transform the `verbose` into `verbosity` ? I will add some aliases for backwards compatibility (if used on Oscar)

**Comment by thofma** [comment]
Good to go?

**Comment by StevellM** [comment]
For me it is.

**Pull request merged**